### ISSUE TITLE
Remove all Drupal 8 @todos from documentation

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#248]
Our documentation contains references to @todos that reference Drupal 8.  We should review these and either:

1. update them to indicate Backdrop 2.x if we agree, or
2. Remove the @todo if we don't want to do the same thing in Backdrop

**Comment by quicksketch** [comment]
PR filed at https://github.com/backdrop/backdrop/pull/644.

For the most part I just removed "D8" or "Drupal 8" from the TODO messages. Two additional situations:

- I removed the intention to remove `$conditions` from `entity_load()`, `user_load()`, `node_load()`, and `taxonomy_term_load()`. We still use this parameter in places. It was intended to be removed because it's not compatible with alternative entity storages (I believe).
- I removed the entire `user_uid_only_optional_to_arg()` function, as it was deprecated even in Drupal 7 and shouldn't be used anywhere already.
